openssl_ocsp_responder/ocsp_responder.py

```diff
@@ -149,14 +161,110 @@ def set_revoked_certificate(self, certificate_path, revocation_time=None):
         self.crl[serial_number]['status'] = 'R'
 
         if revocation_time is None:
-            revocation_time = datetime.datetime.utcnow()
+            revocation_time = datetime.utcnow()
         revocation_time_str = revocation_time.strftime("%y%m%d%H%M%S") + "Z"
         self.crl[serial_number]['revocation_time'] = revocation_time_str
 
     def delete_certificate(self, certificate_path):
         """
         Removes a certificate from the CRL
-        :param certificate_path: path to a PEM certificate file
+        :param str certificate_path: path to a PEM certificate file
         """
         serial_number = self._add_certificate(certificate_path)
         self.crl.pop(serial_number, None)
+
+    def get_status(self, certificate_path, issuer_certificate_path, timeout=None, ocsp_port=None):
+        """
+        Returns the status of a provided certificate by sending a request to the responder
+        :param str certificate_path: path to a PEM certificate file
+        :param str issuer_certificate_path: path to the issuer's certificate PEM file
+        :param int timeout: timeout in seconds for getting a response from the responder
+        :param int ocsp_port: OCSP responder port. used in self-validation of responder, to override reading OCSP URI from certificate
+        :return: the status of the certificate, or None if could not get it (eiter could not send request, or responder
+                 sent an exception status)
+        :rtype: an enum value of cryptography.x509.ocsp.OCSPCertStatus (or None)
+        """
+        request_timeout = self.request_timeout if timeout is None else timeout
+        current_time = datetime.now()
+        status = None
+
+        while datetime.now() < current_time + timedelta(seconds=request_timeout):
+            try:
+                ocsp_response = OCSPResponder.get_cert_ocsp_response(certificate_path, issuer_certificate_path, timeout=0.5, ocsp_port=self.ocsp_port)
+                status = x509.ocsp.load_der_ocsp_response(ocsp_response).certificate_status
+                break
+            except requests.exceptions.Timeout:  # Request timed out
+                pass
+            except requests.exceptions.ConnectionError:  # Responder could not be reached
+                pass
+            except ValueError:  # Response has an unsuccessful status
+                pass
```

Comment:
Maybe get the default logger and log which exception was raised specifically?

Reply:
V